### ISSUE TITLE
ActiveRecord::link(), ActiveRecord::unlink() should return boolean result!

### ISSUE THREAD

**Issue by creocoder** [yiisoft/yii2#10005]
It use `$this->save(false)` inside which can return boolean value and this value is typically checked inside application. For example:

``` php
if (!$model->save(false)) {
    throw new \RuntimeException('Unable to save model.');
}
```

`$model->save(false)` may return false if for example its `beforeSave()` will return false. Currently `link()` and `unlink()` is just unusable in fail tolerance applications because its impossible to check how operation completed. When you do `$this->link()` you can't know will it happen in reality or not. If internal `$this->save(false)` will failed by `beforeSave()` reason `$this->link()` call will silently ignore that and real link will not be happen. The main trouble is: currently impossible to know is real link happen or not.

**Comment by creocoder** [comment]
A more convincing example:

```
class Project extends \yii\db\ActiveRecord
{
    /**
     * @inheritdoc
     */
    public function transactions()
    {
        return [
            self::SCENARIO_DEFAULT => self::OP_ALL,
        ];
    }

    /**
     * @inheritdoc
     */
    public function afterSave($insert, $changedAttributes)
    {
        parent::afterSave($insert, $changedAttributes);

        if ($insert) {
            $collaboration = new Collaboration();
            $collaboration->role = Collaboration::ROLE_YII_CODER;

            if (!$collaboration->link('project', $this)) {
                static::getDb()->transaction->rollBack();
            }
        }
    }
}
```

Currently its impossible, so `link()` is much more less usable than may be.

**Comment by egorio** [comment]
I agree with @creocoder.
I tried to make `if ($order->save() && $item->link($order)) { ...

**Comment by samdark** [comment]
Won't be implemented in 2.0.
